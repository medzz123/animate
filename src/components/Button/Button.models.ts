export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement>;
