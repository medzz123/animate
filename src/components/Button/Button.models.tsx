export type ButtonProps = React.HTMLAttributes<HTMLButtonElement> &
  Partial<ButtonStyles>;

export interface ButtonStyles {
  variant: 'primary' | 'secondary';
  size: 'medium' | 'large';
}
