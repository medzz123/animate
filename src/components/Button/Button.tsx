import React from 'react';

const Button = () => {
  return <button />;
};

export default Button;
