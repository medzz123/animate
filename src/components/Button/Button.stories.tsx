import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Button from './Button';
import { ButtonProps } from './Button.models';

export default {
  title: 'Components/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (storyArguments) => (
  <Button {...storyArguments} />
);

export const ButtonStory = Template.bind({});
ButtonStory.args = {
  children: 'Hello',
};
