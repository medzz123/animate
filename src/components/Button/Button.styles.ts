import { tokens } from '@theme/tokens';
import styled from 'styled-components';

import { ButtonStyles } from './Button.models';

export const ButtonContainer = styled.button.attrs({
  type: 'button',
})<ButtonStyles>`
  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ${(p) => p.theme.accent};

  transition: all ${tokens.transitions.fast} ease 0s;
  font-size: ${(p) =>
    p.size === 'large' ? tokens.sizes[18] : tokens.sizes[12]};

  color: ${(p) => (p.variant === 'primary' ? p.theme.black : p.theme.white)};

  display: inline-flex;
  font-weight: 500;
  padding: ${(p) =>
    p.size === 'large'
      ? `${tokens.sizes[12]}
    ${tokens.sizes[28]}`
      : `${tokens.sizes[4]}
      ${tokens.sizes[12]}`};

  line-height: 1.5;

  &:hover {
    color: ${(p) => (p.variant === 'primary' ? p.theme.white : p.theme.black)};
  }
`;
