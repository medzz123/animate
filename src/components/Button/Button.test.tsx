import React from "react";
import TestRenderer from "react-test-renderer";

import Button from "./Button";

describe("<Button />", () => {
  it("should render without throwing an error and match snapshot", () => {
    const testRenderer = TestRenderer.create(<Button />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
