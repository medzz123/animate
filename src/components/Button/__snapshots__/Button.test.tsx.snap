// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render without throwing an error and match snapshot 1`] = `<button />`;
